src/dag-node.js

```diff
@@ -51,7 +51,7 @@ module.exports = class DAGNode {
     // ensure links are instances of DAGLink
     if (links) {
       links.forEach((l) => {
-        if (l.name && typeof l.toJSON === 'function') {
+        if (l instanceof DAGLink) {
```

Comment:
No! you can not do that (sadly). We explicitly removed this because this will fail if there are two different versions of this module are used. And will give you many false negatives in that case